File: scratch/search_order.py

import json
import os
import requests

def search():
    # Load credentials from the router's .env
    env_path = '/Users/tva/Quantum/project/ohif-orthanc/dicom-router/.env'
    env = {}
    with open(env_path, 'r') as f:
        for line in f:
            if '=' in line:
                k, v = line.strip().split('=', 1)
                env[k] = v

    ORG_ID = env.get('ORG_ID')
    CLIENT_ID = env.get('CLIENT_ID')
    CLIENT_SECRET = env.get('CLIENT_SECRET')
    BASE_URL = env.get('SATUSEHAT_URL')
    ACCESSION = 'test-4444'

    print(f"🔍 Mencari Accession: {ACCESSION} untuk Org: {ORG_ID}")
    
    # 1. Get Token
    auth_url = f"{BASE_URL}/oauth2/v1/accesstoken?grant_type=client_credentials"
    auth_res = requests.post(auth_url, data={
        'client_id': CLIENT_ID,
        'client_secret': CLIENT_SECRET
    }, headers={'Content-Type': 'application/x-www-form-urlencoded'})
    
    auth_data = auth_res.json()
    token = auth_data.get('access_token')
    if not token:
        print(f"❌ Gagal mendapatkan token: {auth_data}")
        return
    print("✅ Auth Sukses.")

    # Daftar system yang mungkin
    systems = [
        f"http://sys-ids.kemkes.go.id/acsn/{ORG_ID}",
        f"https://sys-ids.kemkes.go.id/acsn/{ORG_ID}",
        f"http://sys-ids.kemkes.go.id/servicerequest/{ORG_ID}",
        f"https://sys-ids.kemkes.go.id/servicerequest/{ORG_ID}",
        f"http://sys-ids.kemkes.go.id/acsn",
        f"http://sys-ids.kemkes.go.id/accession/{ORG_ID}",
        f"https://sys-ids.kemkes.go.id/accession/{ORG_ID}",
        f"http://sys-ids.kemkes.go.id/order-id/{ORG_ID}"
    ]

    headers = {'Authorization': f'Bearer {token}'}

    for sys in systems:
        print(f"Testing system: {sys}... ", end='', flush=True)
        try:
            url = f"{BASE_URL}/fhir-r4/v1/ServiceRequest"
            params = {'identifier': f"{sys}|{ACCESSION}"}
            res = requests.get(url, params=params, headers=headers)
            data = res.json()
            
            if data.get('total', 0) > 0:
                print(f"\n\n🎯 KETEMU! System yang benar adalah: {sys}")
                print("Resource identifier structure:")
                print(json.dumps(data['entry'][0]['resource']['identifier'], indent=2))
                return
            else:
                print("❌ Kosong.")
        except Exception as e:
            print(f"⚠️ Error: {str(e)}")

    print("\n🏁 Selesai. Tidak ada yang cocok.")

if __name__ == "__main__":
    search()

File: scratch/search_order.ts

import axios from 'axios';
import fs from 'fs';
import path from 'path';

// Load credentials from the router's .env
const envPath = '/Users/tva/Quantum/project/ohif-orthanc/dicom-router/.env';
const envContent = fs.readFileSync(envPath, 'utf8');
const env = Object.fromEntries(
    envContent.split('\n')
        .filter(line => line && line.includes('='))
        .map(line => line.split('='))
);

const ORG_ID = env.ORG_ID;
const CLIENT_ID = env.CLIENT_ID;
const CLIENT_SECRET = env.CLIENT_SECRET;
const BASE_URL = env.SATUSEHAT_URL;
const ACCESSION = 'test-4444';

async function search() {
    console.log(`Searching for Accession: ${ACCESSION} for Org: ${ORG_ID}`);
    
    // 1. Get Token
    const authUrl = `${BASE_URL}/oauth2/v1/accesstoken?grant_type=client_credentials`;
    const authRes = await axios.post(authUrl, `client_id=${CLIENT_ID}&client_secret=${CLIENT_SECRET}`, {
        headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
    });
    const token = authRes.data.access_token;
    console.log("Auth Success.");

    const systems = [
        `http://sys-ids.kemkes.go.id/acsn/${ORG_ID}`,
        `https://sys-ids.kemkes.go.id/acsn/${ORG_ID}`,
        `http://sys-ids.kemkes.go.id/servicerequest/${ORG_ID}`,
        `https://sys-ids.kemkes.go.id/servicerequest/${ORG_ID}`,
        `http://sys-ids.kemkes.go.id/acsn`,
        `https://sys-ids.kemkes.go.id/acsn`
    ];

    for (const sys of systems) {
        console.log(`Checking system: ${sys}...`);
        try {
            const res = await axios.get(`${BASE_URL}/fhir-r4/v1/ServiceRequest`, {
                params: { identifier: `${sys}|${ACCESSION}` },
                headers: { 'Authorization': `Bearer ${token}` }
            });
            
            if (res.data.total > 0) {
                console.log(`✅ FOUND! System: ${sys}`);
                console.log(JSON.stringify(res.data.entry[0].resource, null, 2));
                return;
            } else {
                console.log(`❌ Not found with this system.`);
            }
        } catch (e) {
            console.log(`⚠️ Error checking ${sys}: ${e.message}`);
        }
    }
    console.log("Finished searching. No results found.");
}

search().catch(console.error);
